src/control/ras/event.go

```diff
@@ -0,0 +1,189 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package ras
+
+/*
+#include "daos_srv/ras.h"
+*/
+import "C"
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/logging"
+)
+
+// EventID describes a given RAS event.
+type EventID uint32
+
+// EventID constant definitions.
+const (
+	EventRankFail   EventID = C.RAS_RANK_FAIL
+	EventRankNoResp EventID = C.RAS_RANK_NO_RESP
+)
+
+func (id EventID) String() string {
+	return C.GoString(C.ras_event_id_enum_to_name(uint32(id)))
+}
+
+// Desc returns a description of the event.
+func (id EventID) Desc() string {
+	return C.GoString(C.ras_event_id_enum_to_msg(uint32(id)))
+}
+
+// Uint32 returns numeric ID of event.
+func (id EventID) Uint32() uint32 {
+	return uint32(id)
+}
+
+// EventSeverityID describes the severity of a given RAS event.
+type EventSeverityID uint32
+
+// EventSeverityID constant definitions.
+const (
+	EventSeverityFatal EventSeverityID = C.RAS_SEV_FATAL
+	EventSeverityWarn  EventSeverityID = C.RAS_SEV_WARN
+	EventSeverityError EventSeverityID = C.RAS_SEV_ERROR
+	EventSeverityInfo  EventSeverityID = C.RAS_SEV_INFO
+)
+
+func (sev EventSeverityID) String() string {
+	return C.GoString(C.ras_event_sev_enum_to_name(uint32(sev)))
+}
+
+// EventTypeID describes the type of a given RAS event.
+type EventTypeID uint32
+
+// EventTypeID constant definitions.
+const (
+	EventTypeStateChange EventTypeID = C.RAS_TYPE_STATE_CHANGE
+	EventTypeInfoOnly    EventTypeID = C.RAS_TYPE_INFO_ONLY
+)
+
+func (typ EventTypeID) String() string {
+	return C.GoString(C.ras_event_type_enum_to_name(uint32(typ)))
+}
+
+// Event describes details of a specific RAS event.
+type Event struct {
+	Name        string `json:"name"`
+	Timestamp   string `json:"timestamp"`
+	Msg         string `json:"msg"`
+	Hostname    string `json:"hostname"`
+	Data        []byte `json:"data"`
+	Eid         uint32 `json:"eid"`
```

Comment:
Should probably call this "ID." Eid is a holiday. :)

Reply:
was following https://wiki.hpdd.intel.com/display/DC/RAS+Events , @johannlombardi happy to rename to ID?

Reply:
no objections , so will change